packages/zoe/src/zoe.js

```diff
@@ -317,36 +361,20 @@ const makeZoe = (additionalEndowments = {}) => {
         return harden(redemptionResult);
       };
 
-      const inviteAmount = inviteIssuer.burn(invite);
-      assert(
-        inviteAmount.extent.length === 1,
-        'only one invite should be redeemed',
-      );
-
-      const {
-        extent: [{ instanceHandle, handle: offerHandle }],
-      } = inviteAmount;
-
-      const { issuers } = instanceTable.get(instanceHandle);
-
       // Promise flow = issuer -> purse -> deposit payment -> escrow receipt
       const paymentDepositedPs = issuers.map((issuer, i) => {
         const issuerRecordP = issuerTable.getPromiseForIssuerRecord(issuer);
         const payoutRule = offerRules.payoutRules[i];
-        const offerPayment = offerPayments[i];
 
         return issuerRecordP.then(({ purse, amountMath }) => {
           if (payoutRule.kind === 'offerAtMost') {
             // We cannot trust these amounts since they come directly
             // from the remote issuer and so we must coerce them.
             return E(purse)
-              .deposit(offerPayment, payoutRule.amount)
+              .deposit(offerPayments[roleNames[i]], payoutRule.amount)
```

Comment:
this makes me want a naming convention for tainted vs. untainted values (ones supplied by a possible attacker vs. already-vetted values). So this should refer to e.g., `safeRoleNames`. I'm not proposing that convention, but @erights might already have one.

Reply:
Proposed convention:

All functions, but especially all functions that can be called from outside the module, should be defensively consistent unless stated otherwise. All others must be marked fragile. Fragile function generally must not be callable from outside the module. Fragile functions that might be called from outside the module need an explicit rationale for why that's safe.

A defensively consistent function, be definition, must be prepared to be called by anything that a fallible caller might call it with. Therefore, all its parameters are suspect. Were we to adopt a naming convention for all variables that might hold suspect values, such as `allegedFoo`, then we would so name all parameter variables, in which case it fails to serve its purpose of alerting the programmer.

Instead, at the top of a function there should be validating code such as

```js
function foo(a, b, allegedC) {
   a = aCoerce(a);
   bValidate(b);
   ...
}
```

where `aCoerce` is a validating coercer for whatever kind of thing `a` is, and `bValidator` is a validator for whatever kind of thing `b` is. Whatever kind of thing `c` is, it is not something we can practically validate in this header. Thus, we name it `allegedC` because, in the code *after* this header, we need to remember to suspect whatever it is bound to.

Reply:
Let me know if I'm summed this up correctly, but it sounds like we are not creating a naming convention for tainted vs. untainted values because that incorrectly assumes that some values are safe. Instead, we should begin a function by either coercing or validating the arguments. If we are able to do that, we don't need to have any special names for the arguments. If we are not able to practically validate or coerce, we use the prefix `alleged`.

Reply:
> because that incorrectly assumes that some values are safe.

Did you mean

"because that incorrectly assumes that some values passed as arguments to non-fragile functions are safe."

Where *non-fragile function* includes all functions not explicitly marked as fragile, i.e., almost all functions.

Other values, for example following validating coercions or those returned by many functions, are safe. This is only about arguments.